chain/chaintest/balance_handler.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chaintest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/codec/codectest"
+	"github.com/ava-labs/hypersdk/consts"
+	"github.com/ava-labs/hypersdk/state/tstate"
+)
+
+type BalanceHandlerFactory func() chain.BalanceHandler
+
+// TestBalanceHandler tests b by requiring that it upholds the invariants
+// described in the BalanceHandler interface.
+func TestBalanceHandler(t *testing.T, ctx context.Context, bf BalanceHandlerFactory) {
```

Comment:
nit: i think a distinct type is a little over-kill here since i don't expect we'll be using this anywhere else.
```suggestion
// TestBalanceHandler tests b by requiring that it upholds the invariants
// described in the BalanceHandler interface.
func TestBalanceHandler(t *testing.T, ctx context.Context, bf func() chain.BalanceHandler) {
```